components/LevelIcon.js

```diff
@@ -0,0 +1,359 @@
+          fill="white"
+        />
+      </>
+    ),
+  },
+];
+
+const COLOR = [
+  // Lv. 5n+1
+  '#FFB600',
+  // Lv. 5n+2
+  '#02D88B',
+  // Lv. 5n+3
+  '#2079F0',
+  // Lv. 5n+4
+  '#966DEE',
+  // Lv. 5n
+  '#FB595A',
+];
+
+const GRADIENT = [
+  // Lv. 5n+1
+  <>
+    <stop stopColor="#FFEA29" />
+    <stop offset="0.0967848" stopColor="#FFEB31" />
+    <stop offset="0.2488" stopColor="#FFED48" />
+    <stop offset="0.4369" stopColor="#FFF16C" />
+    <stop offset="0.6539" stopColor="#FFF69F" />
+    <stop offset="0.8923" stopColor="#FFFCDF" />
+    <stop offset="1" stopColor="white" />
+  </>,
+  // Lv. 5n+2
+  <>
+    <stop stopColor="#4FF695" />
+    <stop offset="0.1113" stopColor="#57F69A" />
+    <stop offset="0.286" stopColor="#6EF8A7" />
+    <stop offset="0.5023" stopColor="#92F9BE" />
+    <stop offset="0.7511" stopColor="#C5FCDC" />
+    <stop offset="1" stopColor="white" />
+  </>,
+  // Lv. 5n+3
+  <>
+    <stop stopColor="#5ED8FF" />
+    <stop offset="0.1186" stopColor="#66DAFF" />
+    <stop offset="0.3047" stopColor="#7DDFFF" />
+    <stop offset="0.5352" stopColor="#A1E8FF" />
+    <stop offset="0.7984" stopColor="#D4F4FF" />
+    <stop offset="1" stopColor="white" />
+  </>,
+  // Lv. 5n+4
+  <>
+    <stop stopColor="#A98AFF" />
+    <stop offset="0.149" stopColor="#AF92FF" />
+    <stop offset="0.3828" stopColor="#BFA9FF" />
+    <stop offset="0.6721" stopColor="#DACDFF" />
+    <stop offset="1" stopColor="white" />
+  </>,
+  // Lv. 5n
+  <>
+    <stop stopColor="#FF7B7B" />
+    <stop offset="0.1366" stopColor="#FF8383" />
+    <stop offset="0.3511" stopColor="#FF9A9A" />
+    <stop offset="0.6166" stopColor="#FFBEBE" />
+    <stop offset="0.9201" stopColor="#FFF1F1" />
+    <stop offset="1" stopColor="white" />
+  </>,
+];
+
+/**
+ *
+ * @param {number} props.level
+ * @param {number} props.width
+ * @param {number} props.height
+ * @param {string?} props.id - The ID for gradient. Must be different for each different <LevelIcon> on one page.
+ */
+function LevelIcon({
+  level,
+  width = 24,
+  height = 24,
+  id = '',
+  style,
+  ...props
+}) {
+  const levelShapeIdx = Math.floor((level - 1) / 5);
+  const colorIdx = (level - 1) % 5;
+  const gradientIdx = `gradient-${id}`;
+
+  // Render nothing for invalid levels or level 0
+  if (!level || !LEVEL_SHAPES[levelShapeIdx]) return null;
+
+  const {
+    width: viewBoxWidth,
+    height: viewBoxHeight,
+    renderContent,
+  } = LEVEL_SHAPES[levelShapeIdx];
+
+  return (
+    <svg
+      width={width}
+      height={height}
+      viewBox={`0 0 ${viewBoxWidth} ${viewBoxHeight}`}
+      preserveAspectRatio="xMidYMid meet"
+      fill="none"
+      xmlns="http://www.w3.org/2000/svg"
+      style={{ ...style, color: COLOR[colorIdx] }}
+      {...(id ? { id: `${id}` } : {})}
+      {...props}
+    >
+      {renderContent(gradientIdx)}
+
```

Comment:
Need to add.
```
<defs>
  put your gradient definition here
</defs>
```

Reply:
Nice catch! `<defs>` is now added and UI does not break after add.
![image](https://user-images.githubusercontent.com/108608/94331203-33421900-fffd-11ea-8118-56624e89d1bd.png)